### PR TITLE
OD-544 [Fix] Fix an issue where date and time pickers couldn't be edited

### PR DESCRIPTION
**Pull request by skrupnyk**
@sofiiakvasnevska @inna-bieshulia

OD-544 https://weboo.atlassian.net/browse/OD-544

Description
Fixed time picker settings

Screenshots/screencasts
https://storyxpress.co/video/kr0fy8y5d7ptv1tkh

Backward compatibility
This change is fully backward compatible.

Reviewers
@upplabs-alex-levchenko

### CODE REVIEW

File: js/components/date.js [hunk 1]
```diff
@@ -43,6 +47,12 @@ Fliplet.FormBuilder.field('date', {
   computed: {
     isWeb: function() {
       return Fliplet.Env.get('platform') === 'web';
+    },
+    isApplyCurrentDateFiled() {
```
Comment (squallstar) [inline js/components/date.js:51]: Filed?

File: js/components/date.js [hunk 2]
```diff
@@ -66,17 +82,24 @@ Fliplet.FormBuilder.field('date', {
...
+    }
+
     this.$emit('_input', this.name, this.value);
     $vm.$v.$reset();
   },
   watch: {
     value: function(val) {
-      if (Fliplet.Env.get('platform') === 'web') {
-        this.datePicker.datepicker('setDate', val);
+      if (!val && this.autofill !== 'empty') {
```
Comment (squallstar) [inline js/components/date.js:101]: Have you verified this works across all platforms? We previously had lots of issues with Upplabs making changes here.

File: js/components/time.js [hunk 3]
```diff
@@ -60,6 +64,14 @@ Fliplet.FormBuilder.field('time', {
       $vm.$v.$reset();
     }
   },
+  computed: {
+    isApplyCurrentDateFiled() {
+      return this.autofill === 'always' || this.autofill === 'default';
+    },
+    isApplySpecificDateField() {
+      return this.autofill === 'specific'
```
Comment (squallstar) [inline js/components/time.js:72]: Missing semicolon

File: js/components/time.js [hunk 4]
```diff
@@ -60,6 +64,14 @@ Fliplet.FormBuilder.field('time', {
       $vm.$v.$reset();
     }
   },
+  computed: {
+    isApplyCurrentDateFiled() {
```
Comment (squallstar) [inline js/components/time.js:68]: Typo in "Filed"

File: js/libs/core.js [hunk 5]
```diff
@@ -181,10 +185,6 @@ Fliplet.FormBuilder = (function() {
         }
       };
 
-      component.methods.onInput = _.debounce(function($event) {
```
Comment (squallstar) [inline js/libs/core.js:184]: why has this been moved?

File: js/libs/core.js [hunk 6]
```diff
@@ -332,6 +332,16 @@ Fliplet.FormBuilder = (function() {
           return;
         }
 
+        if (['flDate', 'flTime'].includes(this._componentName) && this.isApplySpecificDateField && !this.value) {
```
Comment (squallstar) [inline js/libs/core.js:335]: `includes` is not supported in IE11

File: js/libs/core.js [hunk 7]
```diff
@@ -332,6 +332,16 @@ Fliplet.FormBuilder = (function() {
...
+          this.invalidDate = true;
+          this.invalidTime = true;
+
+          return;
```
Comment (squallstar) [inline js/libs/core.js:339]: "Early return" please

File: templates/configurations/time.interface.hbs [hunk 8]
```diff
@@ -18,7 +30,13 @@
...
 </div>
 
-<div class="form-group">
+<div v-if="isApplySpecificDateField" class="form-group" _type="flTime">
+    <label class="control-label">Default value</label>
+    <div v-show="invalidTime" class="text-danger">This field is required*</div>
+    <input type="time" class="form-control" v-model="value">
+</div>
+
+<div v-if="isApplyCurrentDateFiled" class="form-group">
```
Comment (squallstar) [inline templates/configurations/time.interface.hbs:39]: Typo in "Filed"

**Review by squallstar — CHANGES_REQUESTED** [review]
Rejected +1

File: js/libs/core.js (hunk 6 above)
Comment (skrupnyk) [reply js/libs/core.js:335]: > `includes` is not supported in IE11

do we support IE11?

File: js/libs/core.js (hunk 5 above)
Comment (skrupnyk) [reply js/libs/core.js:184]: > why has this been moved?

it has to do with moving another method

File: js/libs/core.js (hunk 6 above)
Comment (squallstar) [reply js/libs/core.js:335]: @sofiiakvasnevska please confirm @skrupnyk has gone through the onboarding material.

File: js/libs/core.js [hunk 9]
```diff
@@ -335,12 +335,10 @@ Fliplet.FormBuilder = (function() {
         if (['flDate', 'flTime'].includes(this._componentName) && this.isApplySpecificDateField && !this.value) {
           this.invalidDate = true;
           this.invalidTime = true;
-
           return;
-        } else {
-          this.invalidDate = false;
-          this.invalidTime = false;
         }
+        this.invalidDate = false;
```
Comment (squallstar) [inline js/libs/core.js:340]: empty line after block conditions

**Comment by inna-bieshulia** [comment]
@sofiiakvasnevska test

File: js/components/date.js (hunk 1 above)
Comment (skrupnyk) [reply js/components/date.js]: fixed

File: js/components/date.js (hunk 2 above)
Comment (skrupnyk) [reply js/components/date.js:101]: this has been tested on all platforms

File: js/components/time.js (hunk 3 above)
Comment (skrupnyk) [reply js/components/time.js:72]: fixed

File: js/components/time.js (hunk 4 above)
Comment (skrupnyk) [reply js/components/time.js]: fixed

File: js/libs/core.js (hunk 6 above)
Comment (skrupnyk) [reply js/libs/core.js]: fixed

File: js/libs/core.js (hunk 9 above)
Comment (skrupnyk) [reply js/libs/core.js:350]: fixed

File: js/libs/core.js [hunk 10]
```diff
@@ -322,6 +322,15 @@ Fliplet.FormBuilder = (function() {
         'props', 'computed'
       ]), component);
 
+      component.methods.includesComponent = function(container, value) {
+        var returnValue = false;
```
Comment (squallstar) [inline js/libs/core.js:326]: Pretty sure this whole function can be rewritten as:

```
return container.indexOf(value) !== -1;
```

Comment (squallstar) [reply js/libs/core.js:326]: Also: your function does not seem to be specific to a component or a container, so why name it as such?

Essentially this is like:

```js
function arrayIncludes(arr, value) {

}
```

**Review by squallstar — CHANGES_REQUESTED** [review]
Rejected +2

File: js/libs/core.js (hunk 10 above)
Comment (skrupnyk) [reply js/libs/core.js:326]: Fixed. I used 'indexOf' in IF block

File: js/libs/core.js [hunk 11]
```diff
@@ -322,6 +322,15 @@ Fliplet.FormBuilder = (function() {
         'props', 'computed'
       ]), component);
 
+      component.methods.includesComponent = function(container, value) {
```
Comment (tonytlwu) [inline js/libs/core.js:325]: Where is this function used?

**Review by tonytlwu — CHANGES_REQUESTED** [review]
Does not meet coding standards.

- Incorrect spacing count with indentation in JS and HTML
- Missing semi-colon
- Missing empty lines before/after a block of variables
- Missing empty lines before/after a `{ }` block 
- Missing empty lines before `return`

Rejected +1

File: js/libs/core.js (hunk 11 above)
Comment (skrupnyk) [reply js/libs/core.js:325]: Fixed. This function was used by me earlier, at the moment I have removed it.

**Review by tonytlwu — CHANGES_REQUESTED** [review]
Does not meet coding standards.

- Incorrect spacing count with indentation in JS and HTML
- Missing semi-colon
- Missing empty lines before/after a block of variables
- Missing empty lines before/after a { } block
- Missing empty lines before return 

Rejected +1

File: templates/configurations/date.interface.hbs [hunk 12]
```diff
@@ -18,7 +30,13 @@
   </div>
 </div>
 
-<div class="form-group">
+<div v-if="isApplySpecificDateField" class="form-group" _type="flDate">
+    <label class="control-label">Default value</label>
```
Comment (tonytlwu) [inline templates/configurations/date.interface.hbs:34]: Check all .hbs files for HTML indentation formatting too.

```suggestion
  <label class="control-label">Default value</label>
```

**Review by tonytlwu — APPROVED** [review]
@sofiiakvasnevska @inna-bieshulia This is ready for testing

Risk: High
Affected: Date and timer pickers in Form component
Verify: Users can manage date and time fields in the app as expected

Squash and merge: Use PR title

**Review by squallstar — APPROVED** [review]

**Pull request merged**